packages/override-tools/src/Cli.ts

```diff
@@ -0,0 +1,210 @@
+ *
+ * @format
+ */
+
+import * as FileRepository from './FileRepository';
+import * as Manifest from './Manifest';
+
+import * as chalk from 'chalk';
+import * as fs from 'fs';
+import * as ora from 'ora';
+import * as os from 'os';
+import * as path from 'path';
+import * as yargs from 'yargs';
+
+import CrossProcessLock from './CrossProcessLock';
+import {GitReactFileRepository} from './GitReactFileRepository';
+import {OverrideFileRepositoryImpl} from './OverrideFileRepository';
+
+doMain(() => {
+  yargs
+    .command(
+      'validate <manifest> [version]',
+      'Verify that overrides are accurate and up-to-date',
+      cmdYargs =>
+        cmdYargs.options({
+          manifest: {type: 'string', describe: 'Path to an override manifest'},
+          version: {type: 'string', describe: 'React Native Version'},
+        }),
+      cmdArgv => validateManifest(cmdArgv.manifest, cmdArgv.version),
+    )
+    .epilogue(
+      'This tool allows maging JavaScript overrides for React Native Windows',
+    )
+    .demandCommand()
+    .recommendCommands()
+    .strict()
+    .showHelpOnFail(false)
+    .wrap(yargs.terminalWidth())
+    .version(false).argv;
+});
+
+/**
+ * Check that the manifest correctly describes files and is up to date
+ */
+async function validateManifest(manifestPath: string, version?: string) {
+  try {
+    await fs.promises.access(manifestPath);
+  } catch {
+    console.error(
+      chalk.red(`Could not find manifest file at path '${manifestPath}'`),
+    );
+    return;
+  }
+
+  const spinner = ora();
+
+  try {
+    spinner.start('Initializing local React Native repo');
+    const [reactRepo, ovrRepo] = await getFileRepos(manifestPath);
+    spinner.succeed();
+
+    let manifest: Manifest.Manifest;
+
+    try {
+      manifest = await Manifest.readFromFile(manifestPath);
+    } catch (ex) {
+      console.error(chalk.red('Could not parse manifest. Is it valid?'));
+      throw ex;
+    }
+
+    spinner.start('Verifying overrides are recorded and up to date');
+    const rnVersion = version || (await getInstalledRNVersion(manifestPath));
+    const validationErrors = await Manifest.validate(
+      manifest,
+      ovrRepo,
+      FileRepository.bindVersion(reactRepo, rnVersion),
+    );
+
+    if (validationErrors.length === 0) {
+      spinner.succeed();
+    } else {
+      spinner.fail();
+      printValidationErrors(validationErrors);
+      process.exitCode = 1;
+    }
+  } catch (ex) {
+    if (spinner.isSpinning) {
+      spinner.fail();
+    }
+    throw ex;
+  }
+}
+
+/**
+ * Create and initialize file repos
+ */
+async function getFileRepos(
+  manifestPath: string,
+): Promise<[GitReactFileRepository, OverrideFileRepositoryImpl]> {
+  const gitDir = os.tmpdir() + path.sep + 'react-native-windows-override-git';
+  const reactRepo = await GitReactFileRepository.createAndInit(gitDir);
+
+  const ovrPath = path.dirname(manifestPath);
+  const ovrFilter = /^.*\.(js|ts|jsx|tsx)$/;
```

Comment:
We're only going to have a single Yoga patch, but I might try to adapt this for native code a bit later.